src/main/java/frc/robot/subsystems/swerve/module/Module.java

```diff
@@ -0,0 +1,310 @@
+	}
+
+	private void setTargetSteerPosition(Rotation2d targetSteerPosition) {
+		steer.applyRequest(steerRequests.position().withSetPoint(targetSteerPosition));
+	}
+
+	public void setTargetVelocity(double targetVelocityMetersPerSecond, boolean isClosedLoop) {
+		if (isClosedLoop) {
+			setTargetClosedLoopVelocity(targetVelocityMetersPerSecond);
+		} else {
+			setTargetOpenLoopVelocity(targetVelocityMetersPerSecond);
+		}
+	}
+
+	public void setTargetClosedLoopVelocity(double targetVelocityMetersPerSecond) {
+		setClosedLoop(true);
+		Rotation2d targetVelocityPerSecond = Conversions.distanceToAngle(targetVelocityMetersPerSecond, constants.wheelDiameterMeters());
+		Rotation2d coupledVelocityPerSecond = ModuleUtils
+			.coupleDriveAngle(targetVelocityPerSecond, steerSignals.velocity().getLatestValue(), constants.couplingRatio());
+		drive.applyRequest(driveRequests.velocity().withSetPoint(coupledVelocityPerSecond));
+	}
+
+	public void setTargetOpenLoopVelocity(double targetVelocityMetersPerSecond) {
+		double voltage = ModuleUtils.velocityToOpenLoopVoltage(
+			targetVelocityMetersPerSecond,
+			steerSignals.velocity().getLatestValue(),
+			constants.couplingRatio(),
+			constants.velocityAt12VoltsPerSecond(),
+			constants.wheelDiameterMeters(),
+			ModuleConstants.VOLTAGE_COMPENSATION_SATURATION
+		);
+		setDriveVoltage(voltage, true);
+	}
+
+
+	/**
+	 * The odometry thread can update itself faster than the main code loop (which is 50 hertz). Instead of using the latest odometry update, the
+	 * accumulated odometry positions since the last loop to get a more accurate position.
+	 *
+	 * @param odometryUpdateIndex the index of the odometry update
+	 * @return the position of the module at the given odometry update index
+	 */
+	public SwerveModulePosition getOdometryPosition(int odometryUpdateIndex) {
+		return new SwerveModulePosition(
+			driveInputs.positionsMeters[odometryUpdateIndex],
+			steerSignals.position().asArray()[odometryUpdateIndex]
+		);
+	}
+
+	public int getNumberOfOdometrySamples() {
+		return Math.min(driveInputs.positionsMeters.length, steerSignals.position().asArray().length);
+	}
+
+	public SwerveModuleState getTargetState() {
+		return targetState;
+	}
+
+	public SwerveModuleState getCurrentState() {
+		return new SwerveModuleState(getDriveVelocityMetersPerSecond(), getSteerPosition());
+	}
+
+	public Rotation2d getDrivePosition() {
+		return driveCouplingInputs.uncoupledPositions[driveCouplingInputs.uncoupledPositions.length - 1];
+	}
+
+	public double getDriveVelocityMetersPerSecond() {
+		return driveInputs.velocityMetersPerSecond;
+	}
+
+	public Rotation2d getSteerPosition() {
+		return steerSignals.position().getLatestValue();
+	}
+
+
+	public boolean isAtTargetVelocity(double speedToleranceMetersPerSecond) {
+		return MathUtil.isNear(getTargetState().speedMetersPerSecond, getDriveVelocityMetersPerSecond(), speedToleranceMetersPerSecond);
+	}
```

Comment:
use the isNear inside the signal inset of the MathUtil one

Reply:
i can't, the signal provide velocity of the motor but here i use the velocity converted to meters